lib/iris/fileformats/_nc_load_rules/actions.py

```diff
@@ -352,8 +387,13 @@ def action_build_dimension_coordinate(engine, providescoord_fact):
             # Get a grid-class from the grid-type
             # i.e. one of latlon/rotated/projected, as for coord_grid_class.
             gridtypes_factlist = engine.fact_list("grid-type")
-            (gridtypes_fact,) = gridtypes_factlist  # only 1 fact
-            (cs_gridtype,) = gridtypes_fact  # fact contains 1 term
+
+            # potentially multiple grid-type facts; find one for CRS varname
+            cs_gridtype = None
+            for fact_cs_name, fact_cs_type in gridtypes_factlist:
+                if fact_cs_name == cs_name:
+                    cs_gridtype = fact_cs_type
```

Comment:
Would `gridtypes_factlist` make more sense as a dictionary? I assume there should be no case where a `fact_cs_name` should appear twice with different `fact_cs_type`s.

Reply:
I think factlists have to be `lists`. From the `FactEntity` docstring:
```
    A Factlist, in turn, is an object with property 'case_specific_facts',
    which is a list of tuples of strings
    (each of which is a 'fact' of the named class).
```

I tried to store the `coordinate_system_mappings` dictionary as a factlist in gridfacts, but just ended up with a list of keys :)